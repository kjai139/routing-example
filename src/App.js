
import { useState } from 'react';
import './App.css';

function App() {

  const [newHeading, setNewHeading] = useState('Old heading')

  const useNewHeading = () => {
    setNewHeading('New heading')
  }

  return (
    <div className="App">
      <h1>Hello from App</h1>
      <p className='newTxt'>{newHeading}</p>
      <button onClick={useNewHeading}>Use new Heading</button>
    </div>
  );
}

export default App;
