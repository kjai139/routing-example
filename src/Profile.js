


const Profile = () => {
    return (
        <h2>Hello from Profile</h2>
    )
}


export {Profile}