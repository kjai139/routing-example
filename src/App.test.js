import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import App from './App';
import { Profile } from './Profile';

test('renders homepage', () => {
  render(<App />);
  const helloApp = screen.getByText(/Hello from App/i);
  expect(helloApp).toBeInTheDocument();
});


test("Profile component", () => {
  
    render(<Profile />);
    const helloProfile = screen.getByText(/Hello from Profile/i)
    expect(helloProfile).toBeInTheDocument()
  
}) 

test("App render", () => {
  const { container } = render(<App />)
  expect(container).toMatchSnapshot()
})


test('new heading after click', () => {
  render(<App />)

  const button = screen.getByRole("button", {name: "Use new Heading" })

  userEvent.click(button)

  expect(screen.getByRole("button").textContent).toMatch(/New heading/i)
})