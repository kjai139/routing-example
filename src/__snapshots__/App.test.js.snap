// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render 1`] = `
<div>
  <div
    class="App"
  >
    <h1>
      Hello from App
    </h1>
    <p
      class="newTxt"
    >
      Old heading
    </p>
    <button>
      Use new Heading
    </button>
  </div>
</div>
`;
